core/src/test/java/io/questdb/test/griffin/engine/table/PageFrameRecordCursorImplFactoryTest.java

```diff
@@ -148,6 +154,366 @@ public void testFactory() throws Exception {
+        final int skip = 0;
+        final long expectedNumOfRows = 10000;
+
+        testFactory_IntervalPartitionFrameCursorFactory(numOfRows, increment, intervalHi, skip, expectedNumOfRows);
+    }
+
+    @Test
+    public void testFactory_IntervalPartitionFrameCursorFactory4_skip() throws Exception {
+        // single partition
+        // interval is the first partition
+        // size is greater than page frame max rows
+        final int numOfRows = 10000;
+        final long increment = 1L;
+        final long intervalHi = 1000000L * 60 * 60 * 24;
+        final int skip = 10;
+        final long expectedNumOfRows = 10000;
+
+        testFactory_IntervalPartitionFrameCursorFactory(numOfRows, increment, intervalHi, skip, expectedNumOfRows);
+    }
+
+    @Test
+    public void testFactory_IntervalPartitionFrameCursorFactory5() throws Exception {
+        // single partition
+        // interval is the first partition
+        // size is less than page frame max rows
+        final int numOfRows = 600;
+        final long increment = 10L;
+        final long intervalHi = 1000000L * 60 * 60 * 24;
+        final int skip = 0;
+        final long expectedNumOfRows = 600;
+
+        testFactory_IntervalPartitionFrameCursorFactory(numOfRows, increment, intervalHi, skip, expectedNumOfRows);
+    }
+
+    @Test
+    public void testFactory_IntervalPartitionFrameCursorFactory5_skip() throws Exception {
+        // single partition
+        // interval is the first partition
+        // size is less than page frame max rows
+        final int numOfRows = 600;
+        final long increment = 10L;
+        final long intervalHi = 1000000L * 60 * 60 * 24;
+        final int skip = 10;
+        final long expectedNumOfRows = 600;
+
+        testFactory_IntervalPartitionFrameCursorFactory(numOfRows, increment, intervalHi, skip, expectedNumOfRows);
+    }
+
+    @Test
+    public void testFactory_IntervalPartitionFrameCursorFactory6() throws Exception {
+        // single partition
+        // interval does not include full partition
+        // size is less than page frame max rows
+        final int numOfRows = 600;
+        final long increment = 100L;
+        final long intervalHi = 1000L * 30;
+        final int skip = 0;
+        final long expectedNumOfRows = 300 + 1;
+
+        testFactory_IntervalPartitionFrameCursorFactory(numOfRows, increment, intervalHi, skip, expectedNumOfRows);
+    }
+
+    @Test
+    public void testFactory_IntervalPartitionFrameCursorFactory6_skip() throws Exception {
+        // single partition
+        // interval does not include full partition
+        // size is less than page frame max rows
+        final int numOfRows = 600;
+        final long increment = 100L;
+        final long intervalHi = 1000L * 30;
+        final int skip = 10;
+        final long expectedNumOfRows = 300 + 1;
+
+        testFactory_IntervalPartitionFrameCursorFactory(numOfRows, increment, intervalHi, skip, expectedNumOfRows);
+    }
+
```

Comment:
Please add tests with timestamp filters that have 2 or more matches per partition. For example, create a table partition by month and use a timestamp filter expression that queries 00:00-12:00 every day.